### DISCUSSION TITLE
Config-flow cannot be loaded error

### DISCUSSION THREAD

**Discussion by VIMVa** [WillCodeForCats/solaredge-modbus-multi#709] [Q&A]
I regularly get this error.
![Screenshot_2024-12-11-10-20-56-61_c3a231c25ed346e59462e84656a70e50](https://github.com/user-attachments/assets/6c81acff-9b7a-475f-b2b8-5de7f64abd15)
and cannot get rid of it, except with a HA restart.
No interruption of data flow as far as I can see.
Ideas?

**Comment by WillCodeForCats** [comment]
I will try and duplicate the issue on my system.

**Comment by WillCodeForCats** [comment]
Can you upload logs that snow any errors and your diagnostics file?

**Reply by VIMVa** [reply]
Will do upon next such error.

**Comment by WillCodeForCats** [comment]
Try upgrading to [Release v3.0.3](https://github.com/WillCodeForCats/solaredge-modbus-multi/releases/tag/v3.0.3) since your error might be related to issue #710.

**Comment by VIMVa** [comment]
Thanks a million for your help trying to trace this.
Will upgrade but I'm still on HA 2024.12.1 (version before the one with which your referenced problem started) yet.
Will keep monitoring and report back.
My error seems (!!!) to appear sometimes (!) when I reboot (e.g. change battery, unplug) an ESPHome device. But that really may just be a wild feeling.

**Reply by WillCodeForCats** [reply]
After looking at this more, it is a mistake I made having requirements incorrectly set without an upper version limit. Pymodbus 3.8.0 introduces breaking API changes that are not backwards compatible. Without an upper limit in the requirements HA will download the latest version of pymodbus which will cause a failure the next time you restart or upgrade Home Assistant.

I've updated the release notes to say [Release v3.0.3](https://github.com/WillCodeForCats/solaredge-modbus-multi/releases/tag/v3.0.3) is a critical update for everyone.

**Comment by VIMVa** [comment]
This error has not appeared any more since then.
